soccer/src/soccer/strategy/agent/position/goalie.cpp

```diff
@@ -35,8 +35,20 @@ Goalie::State Goalie::update_state() {
         return CLEARING;
     }
 
+    rj_geometry::Point robot_position = world_state->get_robot(true, robot_id_).pose.position();
+    double distance_to_ball = robot_position.dist_to(ball_pt);
+    if (latest_state_ == PASSING) {
+        if (!ball_in_box) {
+            return IDLING;
+        }
+    } else if (latest_state_ == RECEIVING) {
+        if (distance_to_ball < 0.1) {
+            return IDLING;
+        }
+    }
```

Comment:
does this goalie passing actually work?

Reply:
Yes, if another agent/robot wants to pass to the goalie (and vice versa) the logic exists and is the same as for passing and receiving for offense and defense.